backend/app/serializers.py

```diff
@@ -53,7 +53,11 @@ def get_analyses(instance):
         for analysis_result in analysis_results:
             name = analysis_result['name']
             result = json.loads(analysis_result['result'])
-            analyses_dict[name] = result
+            if name == "photo_similarity.resnet18_cosine_similarity":
+                analyses_dict[name] = result[:min(10, len(result))]
```

Comment:
Handy thing: slice syntax in Python actually reduces the stop value to the length of the collection if you ask for more values than the length.

So: no need for the call to `min` in this expression -- Python does it for you.